cordformation/src/main/kotlin/net/corda/plugins/Cordapp.kt

```diff
@@ -1,15 +1,28 @@
 package net.corda.plugins
 
 import org.gradle.api.Project
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.Optional
 import java.io.File
 
-open class Cordapp private constructor(val coordinates: String?, val project: Project?) {
+open class Cordapp private constructor(
+    @get:[Optional Input] val coordinates: String?,
```

Comment:
Wow that's some weird looking syntax.

Reply:
But necessary. Otherwise Kotlin will end up annotating the constructor parameters and not the properties! The syntax is telling Kotlin to annotate the property "getters".

Reply:
Yeah, it makes sense. It's just strange looking. It reminds me of PHP somehow.